src/net_processing.cpp

```diff
@@ -842,11 +790,9 @@ void PeerManager::FinalizeNode(const CNode& node, bool& fUpdateConnectionTime) {
     LOCK(cs_main);
     int misbehavior{0};
     {
-        PeerRef peer = GetPeerRef(nodeid);
+        PeerRef peer = RemovePeer(nodeid);
         assert(peer != nullptr);
```

Comment:
This is a little awkward because of the comment in RemovePeer:
```c++
/** Get a shared pointer to the Peer object and remove it from m_peer_map.
 *  May return an empty shared_ptr if the Peer object can't be found. */
```
Might be worth of a comment explaining that _at this point_ the peer is guaranteed to exist, but not afterwards. Then again, that's kinda what the assert means. Shrug.

Reply:
Yeah, I think it's ok for now. Happy to change this if you decide you feel more strongly about it.